### ISSUE TITLE
StackOverflowError at MavenLogReporter.scala:24

### ISSUE THREAD

**Issue by iksnalybok** [SimonJPegg/mvn_scalafmt#44]
at `MavenLogReporter.scala:24`, the error method is calling it-self:

`override def error(file: Path, e: Throwable): Unit = error(file, ScalafmtException(e.getMessage, e))`

**Comment by jozic** [comment]
Found same issue in #43

**Comment by iksnalybok** [comment]
It's indeed about error reporting, but I didn't find that specific error in #43.
The stacktrace I get is:
```
Exception in thread "main" java.lang.StackOverflowError
        at java.lang.Exception.<init>(Exception.java:84)
        at org.scalafmt.dynamic.exceptions.ScalafmtException.<init>(ScalafmtException.scala:6)
        at org.antipathy.mvn_scalafmt.logging.MavenLogReporter.error(MavenLogReporter.scala:24)
        at org.antipathy.mvn_scalafmt.logging.MavenLogReporter.error(MavenLogReporter.scala:24)
        at org.antipathy.mvn_scalafmt.logging.MavenLogReporter.error(MavenLogReporter.scala:24)
        at org.antipathy.mvn_scalafmt.logging.MavenLogReporter.error(MavenLogReporter.scala:24)
        ...
```

**Comment by jozic** [comment]
I mean i saw that it's recursive and fixed it in #43

**Comment by iksnalybok** [comment]
Ok, thanks. I'm thus closing it.
